src/Endatix.Infrastructure/Identity/Authentication/JwtTokenService.cs

```diff
@@ -70,6 +72,48 @@ public TokenDto IssueToken(User forUser, string? forAudience = null)
         return new TokenDto(handler.WriteToken(token), token.ValidTo);
     }
 
+    public Result<long> ValidateAccessToken(string accessToken, bool validateLifetime = true)
+    {
+        var validationParameters = new TokenValidationParameters
+        {
+            IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtOptions.SigningKey)),
+            ValidIssuer = _jwtOptions.Issuer,
+            ValidAudiences = _jwtOptions.Audiences,
+            ValidateIssuer = true,
+            ValidateAudience = true,
+            ValidateLifetime = validateLifetime,
+            ValidateIssuerSigningKey = true,
+            ClockSkew = TimeSpan.FromSeconds(JWT_CLOCK_SKEW_IN_SECONDS)
+        };
+
+        JwtSecurityToken validatedJwtToken;
+        try
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
+            tokenHandler.ValidateToken(accessToken, validationParameters, out var validatedToken);

```

Comment:
Not sure if you have checked this, but the `ValidateTokenAsync` will be easier to translate to our business logic as it throws no exceptions at all - https://learn.microsoft.com/en-us/dotnet/api/system.identitymodel.tokens.jwt.jwtsecuritytokenhandler.validatetokenasync?view=msal-web-dotnet-latest.

> Validates a token. On a validation failure, no exception will be thrown; instead, the exception will be set in the returned TokenValidationResult.Exception property. Callers should always check the TokenValidationResult.IsValid property to verify the validity of the result.

Reply:
Good point, I didn't know with the async method we can get away from handling of an exception, which I definitely didn't like. Thanks.